CMakeLists.txt

```diff
@@ -1,101 +1,103 @@
 cmake_minimum_required(VERSION 3.4.3)
 
+project(OptSched)
+
 option(OPTSCHED_INCLUDE_TESTS "Generate build targets for the OptSched unit tests." ON)
+option(OPTSCHED_ENABLE_AMDGPU "Build the AMDGPU code. Requires that the AMDGPU target is supported." OFF)
+set(OPTSCHED_LIT_ARGS "-sv" CACHE STRING "Arguments to pass to lit")
 set(OPTSCHED_EXTRA_LINK_LIBRARIES "" CACHE STRING "Extra link_libraries to pass to OptSched, ;-separated")
+set(OPTSCHED_EXTRA_INCLUDE_DIRS "" CACHE STRING "Extra include_directories to pass to OptSched, ;-separated")
 # To add OptSched debug defines, e.g.:
 # '-DOPTSCHED_EXTRA_DEFINITIONS=-DIS_DEBUG_DEFS_AND_USES;-DIS_DEBUG_DEF_USE_COUNT'
 set(OPTSCHED_EXTRA_DEFINITIONS "" CACHE STRING "Extra add_definitions to pass to OptSched, ;-separated")
 
-# Exit if attempting to build as a standalone project.
-get_directory_property(hasParent PARENT_DIRECTORY)
-if(NOT hasParent)
-  message(FATAL_ERROR "You cannot build opt-sched as a standalone project. Place opt-sched in the llvm tree under the \"projects\" directory.")
+if(TARGET LLVMCodeGen)
+  set(llvm_subproject TRUE)
+else()
+  set(llvm_subproject FALSE)
+endif()
+
+if(NOT llvm_subproject)
```

Comment:
```
if(NOT llvm_subproject)
  if(OPTSCHED_ENABLE_AMDGPU)
    find_package(LLVM 9.0 REQUIRED CONFIG)
  else()
    find_package(LLVM 6.0 REQUIRED CONFIG)
  endif()
```
Does this mean only LLVM 9.0 or LLVM 6.0 are supported with this build? Me and Paul have been using LLVM 7.0 for the testing with CPU2017.

Reply:
Good point. I just thought of that late last night; I'll have to think about how to support LLVM 7.0 or other versions. It should be fairly straightforward for me to fix this, but I'll have to think about it and test it.

Reply:
OK sounds good. I will test the new build on Monday most likely since I have my work cut out for today. Once LLVM 7 is supported I can also try to merge this build change into my branch and see if it helps with making HIP and CUDA easier to build as well, which would be very nice.

Reply:
The best I could think of is require `-DOPTSCHED_LLVM_VERSION=7.0` if you're configuring with LLVM 7 rather than 6 (or 9 with the AMDGPU).